src/ResourceManager/Resources/Commands.Resources/Models.Authorization/AuthorizationClientExtensions.cs

```diff
@@ -117,7 +118,19 @@ private static IEnumerable<PSRoleAssignment> ToPSRoleAssignments(this IEnumerabl
 
             List<string> objectIds = new List<string>();
             objectIds.AddRange(assignments.Select(r => r.PrincipalId.ToString()));
-            List<PSADObject> adObjects = activeDirectoryClient.GetObjectsByObjectId(objectIds);
+            List<PSADObject> adObjects = null;
+            try
+            {
+                adObjects = activeDirectoryClient.GetObjectsByObjectId(objectIds);
+            }
+            catch (CloudException ce) {
```

Comment:
@darshanhs90 I think we should add a separate function that will check for the below condition:

```cs
public bool IsFooError(CloduException ce)
{
    if (ce.Response != null && ce.Response.StatusCode == HttpStatusCode.Unauthorized &&
        ce.Error != null && ce.Error.Code != null && string.Equals(ce.Error.Code, "Authorization_RequestDenied", StringComparison.OrdinalIgnoreCase))
    {
        return true;
    }
    
    return false
}
```

This allows us to handle this special case and return a custom exception, while allowing the other exceptions to be thrown as they usually would and preserve their error message and content:

```cs
catch (CloudException ex) when (IsFooError(ex))
{
    throw new InvalidOperationException(ProjectResources.InSufficientGraphPermission);
}
```

Reply:
Done

Reply:
@darshanhs90 this is almost correct -- we want to make sure that we're only catching this special type of exception, which is why we'd use the `when` check:

```cs
catch (CloudException ex) when (IsAuthorizationDeniedException(ex))
{
    throw new InvalidOperationException(ProjectResources.InsufficientGraphPermission);
}
```